lib/liquid/tags/doc.rb

```diff
@@ -31,21 +31,26 @@ module Liquid
   class Doc < Block
     NO_UNEXPECTED_ARGS = /\A\s*\z/
 
+    attr_reader :body
+
     def initialize(tag_name, markup, parse_context)
       super
       ensure_valid_markup(tag_name, markup, parse_context)
     end
 
     def parse(tokens)
+      @body = +''
       while (token = tokens.shift)
         tag_name = token =~ BlockBody::FullTokenPossiblyInvalid && Regexp.last_match(2)
 
         raise_nested_doc_error if tag_name == @tag_name
 
         if tag_name == block_delimiter
           parse_context.trim_whitespace = (token[-3] == WhitespaceControl)
+          @body << Regexp.last_match(1) if Regexp.last_match(1) != ""
           return
         end
+        @body << token unless token.empty?
```

Comment:
For my own learning here: what are these two pieces of code actually shoveling into `@body`?

Reply:
Great question! I'm not too familiar with how the tokenizer itself decides what to chunk up, however when we parse Liquid it is parsed into an array of tokens

For example, `{% doc %} Documentation content\n  @param {string} foo - test\n {% enddoc %}` pass the following tokens to `parse`
1) `Documentation content
  @param {string} foo - test`
2) `{% enddoc %}`


To answer your question:
`@body << token unless token.empty?` writes tokens to body, and is doing the bulk of the work

`@body << Regexp.last_match(1) if Regexp.last_match(1) != ""` handles an edge case. When the token being processed looks like `{{ incomplete{% enddoc %}`, we strip the final content before the `{% enddoc %}`. I've added a test for this case as it seems to only be triggered by liquid characters